src/manifest.rs

```diff
@@ -136,3 +144,21 @@ pub fn write_package_json(
 pub fn get_crate_name(path: &str) -> Result<String, Error> {
     Ok(read_cargo_toml(path)?.package.name)
 }
+
+fn has_cdylib(path: &str) -> bool {
+    match read_cargo_toml(path).unwrap().lib {
+        Some(lib) => match lib.crate_type {
+            Some(types) => types.iter().any(|s| s == "cdylib"),
+            _ => false,
+        },
+        _ => false,
+    }
+}
+
+pub fn check_crate_type(path: &str) -> Result<(), Error> {
+    if !has_cdylib(path) {
+        Error::config("crate-type must include cdylib to compile to wasm32-unknown-unknown")
```

Comment:
the existing errors didn't seem appropriate for this use-case, so i added a new one

Reply:
That's totally fine! This function will also need to put a ? at the end of `has_cdylib` once you change it to a Result type as well